### PR TITLE
Add unittest to for the permute CPU implementation

### PR DESCRIPTION
**Pull request by CongMa13**
- compare the permute result with reference.
- test col/row major
- test float and _Float16

### CODE REVIEW

File: test/02_permutation/permutation_cpu_impl_test.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    int              nmodeB = modeB.size();
+
+    std::unordered_map<int, int64_t> extent;
+    extent['h'] = 2;
+    extent['w'] = 3;
+    extent['c'] = 4;
+    extent['n'] = 5;
+
+    std::vector<int64_t> extentA;
+    for(auto mode : modeA)
```
Comment (cgmillette) [inline test/02_permutation/permutation_cpu_impl_test.cpp:50]: For single liner for loops we prefer using scoped {}

File: test/02_permutation/permutation_cpu_impl_test.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                                    &descA,
+                                    modeA.data(),
+                                    bArray.data(),
+                                    &descB,
+                                    modeB.data(),
+                                    typeCompute);
+    return compareEqual(referenceArray.data(), bArray.data(), bArray.size(), 10);
+}
+
+TEST(PermutationCpuImplTest, CompareF32ResultWithNumpy)
```
Comment (cgmillette) [inline test/02_permutation/permutation_cpu_impl_test.cpp]: Please rename to ComparF32ResultWithReference

File: test/02_permutation/permutation_cpu_impl_test.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    hipDataType typeA       = HIP_R_32F;
+    hipDataType typeB       = HIP_R_32F;
+    hipDataType typeCompute = HIP_R_32F;
+
+    auto [result, maxRelativeError]
+        = permuteWithCpu<floatTypeA, floatTypeB, floatTypeCompute>(typeA, typeB, typeCompute);
+    EXPECT_TRUE(result) << "max_relative_error: " << maxRelativeError;
+}
+
+TEST(PermutationCpuImplTest, CompareF16ResultWithNumpy)
```
Comment (cgmillette) [inline test/02_permutation/permutation_cpu_impl_test.cpp]: Please rename to ComparF16ResultWithReference

File: test/02_permutation/permutation_cpu_impl_test.cpp (hunk 1 above)
Comment (CongMa13) [reply test/02_permutation/permutation_cpu_impl_test.cpp:50]: > For single liner for loops we prefer using scoped {}

Fixed

File: test/02_permutation/permutation_cpu_impl_test.cpp (hunk 2 above)
Comment (CongMa13) [reply test/02_permutation/permutation_cpu_impl_test.cpp]: > Please rename to ComparF32ResultWithReference

Fixed

File: test/02_permutation/permutation_cpu_impl_test.cpp (hunk 3 above)
Comment (CongMa13) [reply test/02_permutation/permutation_cpu_impl_test.cpp]: > Please rename to ComparF16ResultWithReference

Fixed

**Review by cgmillette — APPROVED** [review]
LGTM

**Pull request merged**